index.js

```diff
@@ -764,9 +761,18 @@ SendStream.prototype.stream = function stream (path, options) {
     self.onStatError(err)
   })
 
-  // end
-  stream.on('end', function onend () {
-    self.emit('end')
+  this.emit('stream', stream)
+
+  // pipe and end stream manually if set
+  if (options.skipStreamPipe)
+    return;
+
+  stream.pipe(res)
+
+  // response finished, done with the fd
```

Comment:
Why isn't the `onFinished` handler (which will clean up the stream automatically for you) being added when `skipStreamPipe` is true?